### ISSUE TITLE
Fonts are not getting transferred

### ISSUE THREAD

**Issue by andywgarcia** [shelfio/aws-lambda-libreoffice#43]
I am using this to convert a word document to a PDF document in AWS lambda and I have the word document in Times New Roman, but the PDF that is created does not use the Times New Roman font. I am not sure if this is an issue with the packaged libre office or this npm package though.

**Comment by eduardoferrari** [comment]
This packaged version have just arial I think, I had the same problem... to solve, I have copied the windows fonts that I needed and include inside of the .br file. But of course, the package get bigger.

**Comment by andywgarcia** [comment]
> This packaged version have just arial I think, I had the same problem... to solve, I have copied the windows fonts that I needed and include inside of the .br file. But of course, the package get bigger.

@eduardoferrari Do you know if there is a way to do this without modifying the br file? Also, can you list the steps of how to include it in the br file?

**Comment by eduardoferrari** [comment]
Hi @andywgarcia, the file lo.tar.br is compressed using this tool from google: https://github.com/google/brotli/blob/master/c/tools/brotli.md

Inside of this file, you have this folder "instdir/share/fonts/truetype" basically you need to copy your truetype fonts to this folder.

You can also, during the runtime copy the fonts to this folder, since the lambda decompress the file into /temp folder, so in run time, the full path will be "/tmp/instdir/share/fonts/truetype"

You can take the *.ttf files from your Windows machine "c:\Windows\Fonts" to use.

In my view, remaking the lo.tar.br package is the most efficient way to it (in terms of performance) otherwise, you will have an additional step, after the lambda decompress the file, will need take files (from the S3 or from other folder in your packaged) and copy into the correct folder.

I hope I have helped you. :)

**Comment by SanvirDessai** [comment]
Just to add to @eduardoferrari's answer, the files are only extracted when  `convertTo` is called, so that will have to happen before transferring the fonts if you are doing it at runtime.

**Comment by vladgolubev** [comment]
We've [released a major version](https://github.com/shelfio/aws-lambda-libreoffice/releases/tag/v4.0.0) of this package that works with the latest LibreOffice 7.3!

[Check out the docs for a new package here.](https://github.com/shelfio/aws-lambda-libreoffice#usage-for-version-4x-based-on-a-lambda-docker-image)

We're not going to support the previous version of a package that worked with a Lambda Layer and an old 6.4 version of LibreOffice.

Feel free to (re)-open a new issue in regards to the recent package version.


---


Feel free to add new fonts to the base docker image: https://github.com/shelfio/libreoffice-lambda-base-image

**Comment by samuelantonioli** [comment]
Hey,
I found a solution. Put your ttf files into a `fonts/` directory in your repository and use the following Dockerfile:

`Dockerfile`
```
FROM public.ecr.aws/shelf/lambda-libreoffice-base:7.4-node16-x86_64
COPY ./ ${LAMBDA_TASK_ROOT}/
COPY ./fonts ${HOME}/.fonts
# dependency not supplied by lambda-libreoffice-base image
# https://github.com/shelfio/aws-lambda-libreoffice/issues/211
RUN fc-cache && yum install java-1.8.0-openjdk-devel -y
RUN npm install
CMD [ "app.handler" ]
```

What it does is it puts the fonts into the $HOME folder and installs them using `fc-cache`.

`package.json`
```
{
  "dependencies": {
    "@shelf/aws-lambda-libreoffice": "^5.0.0"
  }
}
```

`serverless.yml
```
service: pdfgen-lambda
frameworkVersion: '3'
provider:
  name: aws
  ecr:    
    images:
      appimage:
        path: ./
functions:
  libreoffice:
    url: true
    image:
      name: appimage
```

And then

```bash
sls deploy
sls invoke --function libreoffice
```
